x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -205,7 +207,18 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
 
       inputs = await _compilePackagePolicyInputs(pkgInfo, packagePolicy.vars || {}, inputs);
 
-      elasticsearch = pkgInfo.elasticsearch;
+      if (pkgInfo.type === 'integration') elasticsearch = pkgInfo.elasticsearch;
```

Comment:
Why we do not have `elasticsearch = pkgInfo.elasticsearch` as we have for the update, bulk* method?

Reply:
This was to prevent the new index template fields which input packages can have in the elasticsearch key being applied to package policies, you're right I missed the other methods. 

I've changed the approach to make this more explicit in https://github.com/elastic/kibana/pull/148772/commits/6627977e9221d17415214bab03f30a15e311f330